nixos/modules/services/web-apps/pr-tracker.nix

```diff
@@ -0,0 +1,315 @@
+        defaultText = "/var/lib/pr-tracker/nixpkgs";
+        description = ''
+          The path to the cloned nixpkgs pr-tracker will use.
+
+          ::: {.note}
+          If left as the default value this repo will automatically be cloned before
+          the pr-tracker server starts, otherwise you are responsible for ensuring
+          the directory exists with appropriate ownership and permissions.
+          :::
+        '';
+      };
+
+      remote = mkOption {
+        type = types.str;
+        example = "upstream";
+
+        default = "origin";
+        description = ''
+          The remote name in the repository corresponding to upstream Nixpkgs.
+        '';
+      };
+
+      managedByModule = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether you want this service to manage a clone of the nixpkgs
+          repo in `cfg.dataDir`.
+        '';
+      };
+
+      interval = mkOption {
+        type = types.str;
+        example = "15min";
+
+        default = "30min";
+        description = ''
+          How often to fetch nixpkgs if `cfg.nixpkgsClone.managedByModule` is true.
+        '';
+      };
+    };
+
+    userAgent = mkOption {
+      type = types.str;
+      example = "my pr-tracker";
+
+      description = ''
+        The User-Agent string to use when contacting the GitHub API.
+      '';
+    };
+
+    sourceUrl = mkOption {
+      type = types.str;
+      example = "https://github.com/me/my-pr-tracker-fork";
+
+      default = "https://git.qyliss.net/pr-tracker";
+      description = ''
+        The URL where users can download the program's source code.
+      '';
+    };
+
+    mountPath = mkOption {
+      type = with types; nullOr str;
+      example = "pr-tracker.html";
+
+      default = null;
+      description = ''
+        A "mount" path can be specified, which will be prefixed to all
+        of the server's routes, so that it can be served at a non-root
+        HTTP path.
+      '';
+    };
+
+    user = mkOption {
+      default = "pr-tracker";
+      type = types.str;
+      description = ''
+        User account under which pr-tracker runs.
+
+        ::: {.note}
+        If left as the default value this user will automatically be created
+        on system activation, otherwise you are responsible for ensuring the
+        user exists before the pr-tracker service starts.
+        :::
+      '';
+    };
+
+    group = mkOption {
+      default = "pr-tracker";
+      type = types.str;
+      description = ''
+        Group account under which pr-tracker runs.
+
+        ::: {.note}
+        If left as the default value this user will automatically be created
+        on system activation, otherwise you are responsible for ensuring the
+        user exists before the pr-tracker service starts.
+        :::
+      '';
+    };
+
+    dataDir = mkOption {
+      type = types.path;
+      example = "/home/nixos/git";
+
```

Comment:
That example won't work. I would suggest to drop this option and people wanting to have it somewhere else should use a symlink

Reply:
I could replace it with an option that symlinks the user's nixpkgs path to `/var/lib/pr-tracker` maybe?

Reply:
IMO that would be over engineered. Lets keep this until someone actually needs it.